File: python/OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART/OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART/whole_part/Reservacion.py
from .DatosCliente import DatosCliente
from .DatosCompra  import DatosCompra
from .DatosFuncion import DatosFuncion
        
class Reservacion:
    
    def __init__(self, nombres,
                       apellidoPaterno ,
                       apellidoMaterno ,
                       edad,
                       tipoCliente,
                       numeroTarjeta,
                       fechaVencimiento,
                       asiento,
                       horario,
                       pelicula):
        self.datosCliente = DatosCliente()
        self.datosCompra  = DatosCompra()
        self.datosFuncion = DatosFuncion()
        self.datosCliente.setDatosCliente(nombres,apellidoPaterno,apellidoMaterno,edad,tipoCliente)
        self.datosCompra.setDatosCompra(numeroTarjeta,fechaVencimiento)
        self.datosFuncion.setDatosFuncion(asiento,horario,pelicula)
        
    
    def mostrarReservacion(self):
        print(
        f'''
        Reservación:
        
        Datos cliente:
            Nombres          : {self.datosCliente.nombres}
            Apellido paterno : {self.datosCliente.apellidoPaterno}
            Apellido materno : {self.datosCliente.apellidoMaterno}
            Edad             : {self.datosCliente.edad}
            Tipo de cliente  : {self.datosCliente.tipoCliente}
            
        Datos compra:
            Número de tarjeta               : {self.datosCompra.numeroTarjeta}
            Fecha de vencimiento de tarjeta : {self.datosCompra.fechaVencimiento}
            
        Datos función:
            Asiento  : {self.datosFuncion.asiento}
            Horario  : {self.datosFuncion.horario}
            pelicula : {self.datosFuncion.pelicula}
            
        '''
        )

File: php/OBJECT_ADAPATER_CLASS_ADAPTER_WHOLE_PART/datos/Reservacion.php
<?php

require_once 'DatosCliente.php';
require_once 'DatosFuncion.php';
require_once 'DatosCompra.php';

class Reservacion {

    public $datosCompra;
    public $datosCliente;
    public $datosFuncion;

    public function __construct($nombre, $apellidoP, $apellidoM, $edad, $tipoCliente,
            $asiento, $dia, $mes, $anio, $hora, $pelicula, $numTarjeta, $fecha) {
        $this->datosCliente = new DatosCliente();
        $this->datosCompra = new DatosCompra();
        $this->datosFuncion = new DatosFuncion();

        $this->datosCliente->setDatosCliente($nombre, $apellidoP, $apellidoM, $edad, $tipoCliente);
        $this->datosFuncion->setDatosFuncion($asiento, $hora, $dia, $mes, $anio, $pelicula);
        $this->datosCompra->setDatosCompra($numTarjeta, $fecha);
    }

    public function mostrarReservacion() {
        $informacion = "<h2>Datos de Cliente</h2>";
        $informacion .= "Nombre: " . $this->datosCliente->getNombre();
        $informacion .= "<br/>Apellido Paterno: " . $this->datosCliente->getApellidoPaterno();
        $informacion .= "<br/>Apellido Materno: " . $this->datosCliente->getApellidoMaterno();
        $informacion .= "<br/>Edad: " . $this->datosCliente->getEdad();
        $informacion .= "<br/>Tipo de cliente: " . $this->datosCliente->getTipoCliente() . '<br/>';
        $informacion .= "<h2>Datos de Funcion</h2>";
        $informacion .= "<br/>Pelicula: " . $this->datosFuncion->getPelicula();
        $informacion .= "<br/>Fecha: " . $this->datosFuncion->getDia() .
                "/".$this->datosFuncion->getMes() ."/". $this->datosFuncion->getAnio();
        $informacion .= "<br/>Hora: " . $this->datosFuncion->getHora();
        $informacion .= "<br/>Asiento: " . $this->datosFuncion->getAsiento() . '<br/>';
        $informacion .= "<h2>Datos de Compra</h2>";
        $informacion .= "<br/>Número de tarjeta: " . $this->datosCompra->getNumeroTarjeta();
        $informacion .= "<br/>Fecha: " . $this->datosCompra->getFecha();
        
        return $informacion;
    }

}
